server/services/barcode/CreateBarcodeService.ts

```diff
@@ -42,28 +36,37 @@ export default class CreateBarcodeService {
     font: 'Liberation Sans',
   }
 
-  async generateBarcode(token: string, recipient: Recipient): Promise<BarcodeData> {
-    try {
-      const barcodeValue = await this.createBarcode(token)
-      const barcodeImage = await this.generateBarcodeImage(barcodeValue)
-      return {
-        barcode: barcodeValue,
-        barcodeImageDataUrl: this.generateAddressAndBarcodeImage(barcodeImage, recipient),
-      }
-    } catch (error) {
-      logger.error('Error generating barcode image', error)
-      throw error
-    }
-  }
-
-  private async createBarcode(token: string): Promise<string> {
```

Comment:
The `CreateBarcodeService` class now has 3 public methods on its API:
* generateBarcodeValue
* generateAddressAndBarcodeDataUrlImage
* addBarcodeValuesToRecipients

I'm not entirely convinced that addBarcodeValuesToRecipients lives in this class, but given what it does and how it does it I couldnt think of a neater/easier place to put it.

Reply:
This class already knows what a Recipient is so I think it's fine